src/Compilers/CSharp/Portable/Binder/SwitchBinder_BindPatternSwitch.cs

```diff
@@ -12,44 +12,179 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class SwitchBinder
+    // We use a subclass of SwitchBinder for the pattern-matching switch statement until we have completed
+    // a totally compatible implementation of switch that also accepts pattern-matching constructs.
+    internal partial class PatternSwitchBinder : SwitchBinder
     {
-        private BoundPatternSwitchStatement BindPatternSwitch(SwitchStatementSyntax node, Binder originalBinder, DiagnosticBag diagnostics)
+        // Even though these fields exist in the base class, we record them again here.
+        // Our ultimate intent is to remove the type SwitchBinder and have the fully functional
+        // switch binding implementation present in this class.
```

Comment:
Recorded in #12172

---
In reply to: [64952956](https://github.com/dotnet/roslyn/pull/11485#discussion_r64952956) [](ancestors = 64952956)